File: lib/models/document_tag.dart
class DocumentTag {
  final String? id;
  final String tagName;
  final String userId;
  final String color;
  final DateTime createdAt;

  DocumentTag({
    this.id,
    required this.tagName,
    required this.userId,
    required this.color,
    required this.createdAt,
  });

  Map<String, dynamic> toJson() => {
    if (id != null) '_id': id,
    'tagName': tagName,
    'userId': userId,
    'color': color,
    'createdAt': createdAt.toIso8601String(),
  };

  factory DocumentTag.fromJson(Map<String, dynamic> json) => DocumentTag(
    id: json['_id']?.toString(),
    tagName: json['tagName'] ?? '',
    userId: json['userId'] ?? '',
    color: json['color'] ?? '#FF0000',
    createdAt: json['createdAt'] != null 
        ? DateTime.parse(json['createdAt']) 
        : DateTime.now(),
  );
}

File: backend/models/DocumentTag.js
const mongoose = require('mongoose');

const tagSchema = new mongoose.Schema({
  userId: {
    type: String,
    required: true,
    index: true
  },
  tagName: {
    type: String,
    required: true
  },
  color: {
    type: String,
    default: '#FF0000'
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

tagSchema.index({ userId: 1, tagName: 1 }, { unique: true });

module.exports = mongoose.model('DocumentTag', tagSchema);
